controllers/cloud.redhat.com/providers/sizing/sizing.go

```diff
@@ -0,0 +1,151 @@
+package sizing
+
+//Note on package:
+//I didn't really want to pull this out into its own package
+//I wanted this in database or providers but I ran into dependency cycle problems
+//no matter what I did. So easiest and cleanest solution was just to pull it out
+//that said maybe in the future we can extend sizing out to other stuff in which case
+//a sizing package will be helpful
+
+//Note on naming:
+//Naming is hard. In the context of this API "sizes" are
+//the t shirt sizes (small, medium, etc) whereas "capacities"
+//are the values k8s uses like Gi, M, m, etc. This distinction is
+//important because most of what we're doing here is
+//converting sizes to capacities, so I enforce the distinction
+//strictly. The method names are long, but more importantly, accurate
+
+import (
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+const (
+	//The nature of the beast in this code is we got a lot of magic strings
+	XSMALL string = "x-small"
+	SMALL  string = "small"
+	MEDIUM string = "medium"
+	LARGE  string = "large"
+	XLARGE string = "x-large"
+	//We need to define default sizes because if a ClowdApp doesn't provide
+	//volume or ram/cpu capacities we just get an empty string, so we need
+	//defaults to plug in there
+	DEFAULT_SIZE_VOL     string = XSMALL
+	DEFAULT_SIZE_CPU_RAM string = SMALL
+)
+
+// Public methods
+
+//Get default resource requirement requests and limits
+func GetDefaultResourceRequirements() core.ResourceRequirements {
+	return GetResourceRequirementsForSize(GetDefaultSizeCPURAM())
+}
+
+//Gets the default size for CPU and RAM
+func GetDefaultSizeCPURAM() string {
+	return DEFAULT_SIZE_CPU_RAM
+}
+
+//Gets the default vol size
+func GetDefaultSizeVol() string {
+	return DEFAULT_SIZE_VOL
+}
+
+//Get the default volume capacity
+func GetDefaultVolCapacity() string {
+	return getVolSizeToCapacityMap()[GetDefaultSizeVol()]
+}
+
+//Get resource requirements - request and limits - for a given size
+func GetResourceRequirementsForSize(tShirtSize string) core.ResourceRequirements {
+	requestSize := useDefaultIfEmptySize(tShirtSize, GetDefaultSizeCPURAM())
+	cpu := getCPUSizeToCapacityMap()
```

Comment:
Why return the entire map here, when the function could accept an arg and be used directly on line 67, 68 etc?